unskript-ctl/unskript_ctl_list.py

```diff
@@ -0,0 +1,280 @@
+    incomplete_creds = []
+
+    creds_files = os.environ.get('HOME').strip('"') + CREDENTIAL_DIR + '/*.json'
+    list_of_creds = glob.glob(creds_files)
+
+    for creds in list_of_creds:
+        with open(creds, 'r') as f:
+            c_data = json.load(f)
+
+            connector_type = c_data.get('metadata').get('type')
+            connector_name = c_data.get('metadata').get('name')
+
+            if c_data.get('metadata') and c_data.get('metadata').get('connectorData') != "{}":
+                active_creds.append((connector_type, connector_name))
+            else:
+                incomplete_creds.append((connector_type, connector_name))
+
+    combined_list = active_creds + incomplete_creds
+
+    headers = ["#", "Connector Type", "Connector Name", "Status"]
+    table_data = [headers]
+
+    for index, (ctype, cname) in enumerate(combined_list, start=1):
+        status = "Active" if index <= len(active_creds) else "Incomplete"
+        table_data.append([index, ctype, cname, status])
+
+    print(tabulate(table_data, headers='firstrow', tablefmt='fancy_grid'))
+
+def list_checks_by_connector():
+    """list_checks_by_connector This function reads the ZoDB and displays the
+       checks by a given connector. connector_name can be `all` in that case
+       all the checks across connectors performed and displayed.
+
+       :rtype: None
+    """
+    parser = ArgumentParser(description='-c | --checks')
+    parser.add_argument('-l',
+                        '--list',
+                        action="store_true",
+                        help="List Option")
+    parser.add_argument('-c',
+                        '--checks',
+                        action='store_true',
+                        help="List Checks")
+    parser.add_argument('-a',
+                        '--all',
+                        help="List all Checks across connectors in the system",
+                        action="store_true")
+    parser.add_argument('-t',
+                        '--type',
+                        choices=CONNECTOR_LIST,
+                        help="Type of connector for which the checks should be shown")
+
+    args = parser.parse_args()
+
+    if len(sys.argv) <= 2:
+        parser.print_help()
+        sys.exit(0)
+
+
+    if args.all is True:
+        connector_name = 'all'
+    elif args.type not in ('', None):
+        connector_name = args.type
+    else:
+        connector_name = 'all'
+
+    list_connector_table = [
+        [TBL_HDR_LIST_CHKS_CONNECTOR, TBL_HDR_CHKS_NAME, TBL_HDR_CHKS_FN]]
+    for l in get_checks_by_connector(connector_name):
+        list_connector_table.append(l)
+
+    print("")
+    print(tabulate(list_connector_table, headers='firstrow', tablefmt='fancy_grid'))
+    print("")
+
+def list_main():
+    lp = ArgumentParser()
+    lp.add_argument('-l',
+                    '--list',
+                    action='store_true',
+                    help='List Options')
+    lp.add_argument('-r',
```

Comment:
this doesnt make sense, -r, -c, --failed-checks are all options after -l, so why are these arguments defined at this level? Look at how i do it hierarchical in add_creds.py